src/library-authoring/LibraryCollections.tsx

```diff
@@ -1,14 +1,62 @@
-import React from 'react';
-import { FormattedMessage } from '@edx/frontend-platform/i18n';
-
-import messages from './messages';
-
-const LibraryCollections = () => (
-  <div className="d-flex my-6 justify-content-center">
-    <FormattedMessage
-      {...messages.collectionsTempPlaceholder}
-    />
-  </div>
-);
+import { CardGrid } from '@openedx/paragon';
+
+import { useLoadOnScroll, useSearchContext } from '../search-manager';
+import { NoComponents, NoSearchResults } from './EmptyStates';
+import CollectionCard from './components/CollectionCard';
+import { LIBRARY_SECTION_PREVIEW_LIMIT } from './components/LibrarySection';
+
+type LibraryCollectionsProps = {
+  variant: 'full' | 'preview',
+};
+
+/**
+ * Library Collections to show collections grid
+ *
+ * Use style to:
+ *   - 'full': Show all collections with Infinite scroll pagination.
+ *   - 'preview': Show first 4 collections without pagination.
+ */
+const LibraryCollections = ({ variant }: LibraryCollectionsProps) => {
+  const {
+    collectionHits,
+    totalCollectionHits,
+    isFetchingNextPage,
+    hasNextPage,
+    fetchNextPage,
+    isFiltered,
+  } = useSearchContext();
+
+  const collectionList = variant === 'preview' ? collectionHits.slice(0, LIBRARY_SECTION_PREVIEW_LIMIT) : collectionHits;
+
+  useLoadOnScroll(
+    hasNextPage,
+    isFetchingNextPage,
+    fetchNextPage,
+    variant === 'full',
+  );
+
+  if (totalCollectionHits === 0) {
+    return isFiltered ? <NoSearchResults searchType="collection" /> : <NoComponents searchType="collection" />;
+  }
+
+  return (
+    <CardGrid
```

Comment:
We are changing the card/grid style here: https://github.com/openedx/frontend-app-course-authoring/pull/1258

Just a note to adjust here if the other PR is merged first.